packages/react-vapor/src/components/slider/MiddleSlider.tsx

```diff
@@ -0,0 +1,133 @@
+import 'rc-slider/assets/index.css';
+
+import {Range, SliderProps} from 'rc-slider';
+import * as React from 'react';
+import {connect} from 'react-redux';
+
+import {IDispatch} from '../../utils/ReduxUtils';
+import {SliderActions} from './SliderActions';
+import SliderHandle from './SliderHandle';
+import {
+    convertInitialValuetoRangeValue,
+    getComputedRangeValue,
+    getCrossingPoint,
+    getValuesPositionOnRange,
+    handleIsAtCrossingPoint,
+    propsValidator,
+    valuesPositionOnRange,
+} from './SliderUtils';
+
+export interface MiddleSliderOwnProps extends SliderProps {
+    id: string;
+    enabled?: boolean;
+    hasTooltip?: boolean;
+    initialValue?: number;
+    marks?: {[key: number]: string};
+    range: [number, number];
+    step?: number;
+    tabIndex?: number[];
+    onChange?: (rangeOutputValue: number) => any;
+    customTooltip?: (value: any) => JSX.Element;
+}
+
+export const mapDispatchToProps = (dispatch: IDispatch, ownProps: MiddleSliderOwnProps) => ({
+    setOutputValue: (value: number) => dispatch(SliderActions.setValue(ownProps.id, value)),
+});
+
+const MiddleSliderDisconnected: React.FunctionComponent<MiddleSliderOwnProps &
+    ReturnType<typeof mapDispatchToProps>> = ({
+    enabled = true,
+    customTooltip,
+    hasTooltip,
+    initialValue,
+    marks,
+    range,
+    setOutputValue,
+    step,
+    onChange,
+}) => {
+    const crossingPoint = getCrossingPoint(range);
+    const [highRange, setHighRange] = React.useState(crossingPoint);
+    const [lowRange, setLowRange] = React.useState(crossingPoint);
+    const rangeOutputValue = getComputedRangeValue(lowRange, highRange, range, crossingPoint);
+
+    if (onChange) {
+        onChange(rangeOutputValue);
+    }
+
+    React.useEffect(() => {
+        propsValidator(range, initialValue);
+        if (initialValue) {
+            const initialRangeValue: number = convertInitialValuetoRangeValue(range, initialValue);
+            setHandlePosition([initialRangeValue, initialRangeValue]);
+        }
+    }, []);
```

Comment:
```suggestion
    }, [range, initialValue]);
```

Reply:
I just want this to execute when the component mounts. So I don't think this is relevant.

Reply:
https://overreacted.io/a-complete-guide-to-useeffect/#dont-lie-to-react-about-dependencies
Written by the inventor of the react hooks :arrow_up: :slightly_smiling_face:

Reply:
ok you win 🤺